x-pack/filebeat/tests/system/test_http_endpoint.py

```diff
@@ -308,6 +308,50 @@ def test_http_endpoint_invalid_hmac(self):
         assert r.status_code == 401
         self.assertRegex(r.json()['message'], 'invalid HMAC signature')
 
+    def test_http_endpoint_preserve_original_event(self):
+        """
+        Test http_endpoint input with correct auth header and secret.
+        """
+        options = """
+  preserve_original_event: true
+"""
+        self.get_config(options)
+        filebeat = self.start_beat()
+        self.wait_until(lambda: self.log_contains("Starting HTTP server on {}:{}".format(self.host, self.port)))
+
+        message = "somerandommessage"
+        payload = {self.prefix: message}
+        headers = {"Content-Type": "application/json"}
+        r = requests.post(self.url, headers=headers, data=json.dumps(payload))
```

Comment:
You might even want to make the `json.dumps` pretty with an `indent` so that you can verify that you are getting the exact bytes back and not something that was reconstructed (assuming that's the requirement you are fulfilling).